### PR TITLE
Update DG appendix for consultations

### PR DESCRIPTION
**Pull request by wesho1107**

### CODE REVIEW

File: docs/DeveloperGuide.md [hunk 1]
```diff
@@ -1423,6 +1459,53 @@ testers are expected to do more *exploratory* testing.
 
 1. _{ more test cases …​ }_
 
+### Creating a consultation
+
+1. Creating a consultation with specified date, time and student(s).
+
+    1. Prerequisites: Date and time inputs must be valid and student names must exist in address book. There must not be a duplicate consultations (same date, time and students).
```
Comment (Originalidk) [inline docs/DeveloperGuide.md]: It should be "... a duplicate consultation"

File: docs/DeveloperGuide.md [hunk 2]
```diff
@@ -357,37 +357,40 @@ assertEquals(testFromObjects, testFromString);
...
-
 * **Alternative 2:** Use ArrayList<Person> to keep track of students.
     * Cons: We must ensure there are no duplicates with additional checks.
 
-**Aspect: Adding students to a new or existing consultation:**
+* **Alternative 3 (current choice):** Use a `StudentSet` class to keep track of students.
+    * Pros: Better abstraction and easier maintainability.
+    * Cons: Performance overhead related to creation and manipulation of new class.
+
+**Aspect: Adding or removing students to or from a consultation:**
```
Comment (Originalidk) [inline docs/DeveloperGuide.md]: Just a nit, but kept the colon here but removed it from other parts.

File: src/main/java/seedu/address/logic/parser/ParserUtil.java [hunk 3]
```diff
@@ -43,8 +43,8 @@
 public class ParserUtil {
 
     public static final String MESSAGE_INVALID_INDEX = "Index not a non-zero unsigned integer.";
-    public static final String MESSAGE_INVALID_DATE = "Date needs to be in the format dd/MM/yyyy.";
-    public static final String MESSAGE_INVALID_TIME = "Time needs to be in the format HH:mm.";
+    public static final String MESSAGE_INVALID_DATE = "Date needs to be a valid date and in the format dd/MM/yyyy";
+    public static final String MESSAGE_INVALID_TIME = "Time needs to be valid time and in the format HH:mm.";
```
Comment (Originalidk) [inline src/main/java/seedu/address/logic/parser/ParserUtil.java]: Should add an "a" in front of valid.

**Review by Originalidk — APPROVED** [review]
Other than a few nits, LGTM :)

File: docs/DeveloperGuide.md (hunk 1 above)
Comment (wesho1107) [reply docs/DeveloperGuide.md]: I think duplicate consultations.

Comment (wesho1107) [reply docs/DeveloperGuide.md]: No, you're right!

**Pull request merged**